File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/answer/ErrorCode.java
package com.github.dedis.popstellar.model.network.answer;

import java.util.Objects;

/**
 * Error of a failed request
 */
public final class ErrorCode {

  private final int code;
  private final String description;

  /**
   * Constructor of an ErrorCode
   *
   * @param code        the code of the error, as an integer
   * @param description the description of the error
   */
  public ErrorCode(int code, String description) {
    this.code = code;
    this.description = description;
  }

  /**
   * Returns the code of the error.
   */
  public int getCode() {
    return code;
  }

  /**
   * Returns the description of the error.
   */
  public String getDescription() {
    return description;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ErrorCode that = (ErrorCode) o;
    return getCode() == that.getCode() && Objects.equals(getDescription(), that.getDescription());
  }

  @Override
  public int hashCode() {
    return Objects.hash(getCode(), getDescription());
  }

  @Override
  public String toString() {
    return "ErrorCode{code=" + code + ", description='" + description + "'}";
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/answer/ErrorCode.kt
package com.github.dedis.popstellar.model.network.answer

import java.util.Objects

/** Error of a failed request  */
class ErrorCode
/**
 * Constructor of an ErrorCode
 *
 * @param code the code of the error, as an integer
 * @param description the description of the error
 */(
        /** Returns the code of the error.  */
        @JvmField val code: Int,
        /** Returns the description of the error.  */
        @JvmField val description: String) {

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as ErrorCode
        return this.code == that.code && description == that.description
    }

    override fun hashCode(): Int {
        return Objects.hash(this.code, description)
    }

    override fun toString(): String {
        return "ErrorCode{code=$code, description='$description'}"
    }
}